core/src/apps/management/apply_settings.py

```diff
@@ -12,6 +12,21 @@
     from trezor.messages.ApplySettings import ApplySettings, EnumTypeSafetyCheckLevel
 
 
+def validate_homescreen(homescreen: bytes) -> None:
+    if homescreen == b"":
+        return
+
+    if len(homescreen) > storage.device.HOMESCREEN_MAXSIZE:
+        raise wire.DataError("Homescreen is too complex")
```

Comment:
These wire errors are propagated to the host so it's good to be more expressive. I know this was there already but I suggest to improve it. Maybe `Homescreen image is too large. Maximum size is XY.`?

Reply:
Fixed 277cb136.